oak_restricted_kernel_sdk/src/dice.rs

```diff
@@ -24,6 +24,45 @@ use zerocopy::{AsBytes, FromZeroes};
 lazy_static::lazy_static! {
     static ref DICE_WRAPPER: anyhow::Result<DiceWrapper> = {
         let dice_data = get_restricted_kernel_dice_data()?;
+        let dice_wrapper = dice_data.try_into()?;
+        Ok(dice_wrapper)
+    };
+}
+
+#[cfg(not(feature = "mock_attestion"))]
+fn get_restricted_kernel_dice_data() -> anyhow::Result<RestrictedKernelDiceData> {
+    let mut result = RestrictedKernelDiceData::new_zeroed();
+    let buffer = result.as_bytes_mut();
+    let len = read(DICE_DATA_FD, buffer).map_err(|err| anyhow::anyhow!("read failure: {err}"))?;
+    if len != buffer.len() {
+        anyhow::bail!("invalid dice data size");
+    }
+    Ok(result)
+}
+
+#[cfg(feature = "mock_attestion")]
```

Comment:
Personally I still would prefer a different function with a different name, and make the caller choose which one to call. Hiding the switch behind a feature does not remove the need for callers to think about it, and makes it impossible to actually control it (e.g. it is impossible to enforce that this feature must be disabled, because as long as any transitive dep enables it, it will be enabled anyways).

Reply:
> makes it impossible to actually control it (e.g. it is impossible to enforce that this feature must be disabled, because as long as any transitive dep enables it, it will be enabled anyways

that's a very good point. 

I initially used just the feature to stick as close to the design doc as possible. — Updated to require explicit initialization. I'm still convinced that the ideal interface would be one of traits. It'd solve this much more elegantly.